File: inheritance.py
class Parent:
    def __init__(self, parent_name):
        print("Parent constructor", parent_name, sep=" - ")
        self.pname = parent_name

    def write_name(self):
        print(self.pname)


class Child(Parent):  # Parent sınıfı Child sınıfı tarafından kalıtılıyor
    def __init__(self, name):
        print("Child constructor", name, sep=" - ")
        Parent.__init__(self, name)
        #super().__init__(name) # bir üstteki satır yerine bu satır da kullanılabilir.


c = Child("Mertt")
c.write_name()

File: inheritance.cs
using System;

public class Example
{
   public static void Main()
   {
      var child = new Child("Mertt");
	   child.writeName();
   }
}

public class Parent{
	public string Name;
	public Parent(string name){
		Console.WriteLine("Parent constructor - " + name);
		Name = name;
	}
	public void writeName(){
		Console.WriteLine(Name);
	}
}

public class Child : Parent {
	public Child(string name):base(name){
		Console.WriteLine("Child constructor - "+ name);
	}
}
